zerver/lib/bugdown/__init__.py

```diff
@@ -994,6 +1022,7 @@ def extendMarkdown(self, md, md_globals):
             ModalLink(r'!modal_link\((?P<relative_url>[^)]*), (?P<text>[^)]*)\)'),
             '>avatar')
         md.inlinePatterns.add('usermention', UserMentionPattern(mention.find_mentions), '>backtick')
+        md.inlinePatterns.add('stream', StreamPattern(r'(?<![^\s\'\"\(,:<])#(?:\*\*([^\*]+)\*\*|(\w+))'), '>backtick')
```

Comment:
We prefer to use named groups in our regular expressions (i.e .the `(?P<text>` thing above); can you switch to that style?  Also, can you comment this in more the style of `paren_group` below?  It's more readable that way :).